android/src/main/java/com/SplashPad/export_plugin/ExportPlugin.java

```diff
@@ -102,53 +88,60 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
   @Override
   public boolean onActivityResult(int requestCode, int resultCode, Intent intent) {
 
-    //GUARD
+    if (BuildConfig.DEBUG) {
+        Log.d("ExportPlugin", "onActivityResult: intent = " + intent);
+    }
+
+    //GUARD : only handle create file requests
     if (requestCode != CREATE_FILE) return false;
 
-    if (BuildConfig.DEBUG) {
-      Log.d("ExportPlugin", "onActivityResult: intent = " + intent);
+    if (resultCode == RESULT_CANCELED) {
+        // If we get a cancelled result, we should reset the result and path
+        result.error("result cancelled", null, null);
+        result = null;
+        path = null;
+        return false;
     }
 
-    try {
-      if (intent != null && intent.getData() != null) {
-        writeFile(intent.getData());
-        result.success("Successfully saved to local storage");
-      } else {
-        result.error(null, "No directory chosen", null);
-      }
-    } catch (final Throwable e) {
-      result.error(null,"Failed to save file", e);
-    } finally {
-      result = null;
-      path = null;
+    if (resultCode == RESULT_OK) {
+        // GUARD : to make sure this block does not run even after cancelled
+        // result is null only in two conditions, a success RESULT_OK return, or a RESULT_CANCELLED return
+        if (result == null) {
+            return false;
+        }
+
```

Comment:
I worry that this will cause us to lose the error logging. It is still unknown why result.error is being called when null, my guess is that onActivityResult is firing multiple times